src/StaticWebAssetsSdk/Tasks/DefineStaticWebAssets.Cache.cs

```diff
@@ -0,0 +1,217 @@
+
+    internal class DefineStaticWebAssetsCache
+    {
+        private readonly List<ITaskItem> _assets = [];
+        private readonly List<ITaskItem> _copyCandidates = [];
+        private string? _manifestPath;
+        private IDictionary<string, ITaskItem> _inputHashes = new Dictionary<string, ITaskItem>();
+        private bool _cacheUpToDate;
+        private TaskLoggingHelper? _log;
+
+        public DefineStaticWebAssetsCache() { }
+
+        internal DefineStaticWebAssetsCache(TaskLoggingHelper log, string? manifestPath) : this()
+            => (_log, _manifestPath) = (log, manifestPath);
+
+        // Inputs for the cache
+        public byte[] GlobalPropertiesHash { get; set; } = [];
+        public byte[] FingerprintPatternsHash { get; set; } = [];
+        public byte[] PropertyOverridesHash { get; set; } = [];
+        public HashSet<string> InputHashes { get; set; } = [];
+
+        // Outputs for the cache
+        public Dictionary<string, StaticWebAsset> CachedAssets { get; set; } = [];
+        public Dictionary<string, CopyCandidate> CachedCopyCandidates { get; set; } = [];
+
+        internal static DefineStaticWebAssetsCache ReadOrCreateCache(TaskLoggingHelper log, string manifestPath)
+        {
+            if (manifestPath != null && File.Exists(manifestPath))
+            {
+                using var existingManifestFile = File.OpenRead(manifestPath);
+                var cache = JsonSerializer.Deserialize(existingManifestFile, DefineStaticWebAssetsSerializerContext.Default.DefineStaticWebAssetsCache);
+                if (cache == null)
+                {
+                    throw new InvalidOperationException($"Failed to deserialize cache from {manifestPath}");
+                }
+                cache.SetPathAndLogger(manifestPath, log);
+                return cache;
+            }
+            else
+            {
+                return new DefineStaticWebAssetsCache(log, manifestPath);
+            }
+        }
+
+        internal void WriteCacheManifest()
+        {
+            if (_manifestPath != null)
+            {
+                using var manifestFile = File.OpenWrite(_manifestPath);
+                manifestFile.SetLength(0);
+                JsonSerializer.Serialize(manifestFile, this, DefineStaticWebAssetsSerializerContext.Default.DefineStaticWebAssetsCache);
+            }
+        }
+
+        internal void AppendAsset(string hash, StaticWebAsset asset, ITaskItem item)
+        {
+            asset.AssetKind = item.GetMetadata(nameof(StaticWebAsset.AssetKind));
+            _assets.Add(item);
+            CachedAssets[hash] = asset;
+        }
+
+        internal void AppendCopyCandidate(string hash, string identity, string targetPath)
+        {
+            var copyCandidate = new CopyCandidate(identity, targetPath);
+            _copyCandidates.Add(copyCandidate.ToTaskItem());
+            CachedCopyCandidates[hash] = copyCandidate;
+        }
+
+        internal void PrepareForProcessing(
```

Comment:
Do you think this would be clearer if it was called `Update()`? I didn't have any idea what this method did before reading the implementation.